docs/HowToGuides.md

```diff
@@ -1,5 +1,78 @@
 # How To ...
 
+### Create a multi-step PR
+
+> Ask a programmer to review 10 lines of code, he'll find 10 issues. Ask him to do 500 lines and he'll say it 
+> looks good. _[[source](https://twitter.com/girayozil/status/306836785739210752)]_
+
+Big PRs (i.e. PRs touching more than 100 lines) are hard to review. Using the _multi-step PR_ format allows the 
+review to be done incrementally.
+
+A multi-step PR consists a series of commits corresponding to a sequence of logical steps, 
+each step advancing the code base towards the end goal of the PR. The build must pass at each step. 
```

Comment:
Some suggested additions to the criteria of what makes a good multi-step PR. (Need to tweak the wording to make them read better as required)

* Each commit contains a single logical change, and this change must stand on its own. (or, you can word it as: each commit has a single responsibility, and this responsibility must be fully carried out) (i.e. in AddressBook, a commit that claims it is implementing an `EditCommand` cannot have its `EditCommand::execute()` method throw a "Not Implemented" error, since this means that the commit did not *fully* implement the `EditCommand` which is at odds with what it claims to do. We can't use this example in a document that's supposed to be generic though :-P)

* The series of commits in the PR are ordered in a bottom-up fashion, each commit building on top of each other towards the end goal of the PR.

* The build passes at each commit of the PR.

* Each commit has a detailed commit message which explains the context and rationale behind the commit. See the writing commit messages doc for more details.